client/pluginmanager/group.go

```diff
@@ -65,14 +65,12 @@ func (m *PluginGroup) WaitForFirstFingerprint(ctx context.Context) (<-chan struc
 		go func() {
 			defer wg.Done()
 			logger.Debug("waiting on plugin manager initial fingerprint")
+			<-manager.WaitForFirstFingerprint(ctx)
```

Comment:
Does `WaitForFirstFingerprint` have a side-effect? I'd have thought we'd want something more like:

```go
select {
	case <- manager.WaitForFirstFingerprint(ctx):
		logger.Debug("finished plugin manager initial fingerprint")
	case <-ctx.Done():
		logger.Warn("timeout waiting for plugin manager to be ready")
}
```

Reply:
My goal here was not to change the behavior; are you saying you think it has changed, or that we should change it further?

Reply:
I was saying that as long as we're touching this code we should introduce this change and fix what looks like a minor bug. But that's a good point... maybe that should be under separate PR so that it's not getting lost in this linter PR.